apis/Google.Cloud.Datastore.V1/Google.Cloud.Datastore.V1/DatastoreDbBuilder.cs

```diff
@@ -140,29 +85,37 @@ void ApplyEmulatorConfiguration()
         protected override GrpcAdapter DefaultGrpcAdapter =>
             throw new InvalidOperationException($"This property should never execute in {nameof(DatastoreDbBuilder)}");
 
-        private class EmulatorConfiguration
+        /// <summary>
+        /// Common code for handling project ID, namespace ID and emulator detection.
+        /// </summary>
+        private ConfiguredBuilder PrepareBuilder()
         {
-            internal const string EmulatorHostVariable = "DATASTORE_EMULATOR_HOST";
-            internal const string EmulatorProjectVariable = "DATASTORE_PROJECT_ID";
-            internal string Endpoint { get; }
-            internal string ProjectId { get; }
-
-            private EmulatorConfiguration(string endpoint, string projectId)
+            var emulatorBuilder = (DatastoreDbBuilder) MaybeUseEmulator();
+            if (emulatorBuilder is object)
             {
-                Endpoint = endpoint;
-                ProjectId = projectId;
+                return emulatorBuilder.PrepareBuilder();
             }
+            GaxPreconditions.CheckState(!string.IsNullOrEmpty(ProjectId), "The project ID must be configured");
 
-            internal static EmulatorConfiguration FromEnvironment()
-            {
-                // Note: we treat present-but-empty environment variables as if they were absent.
-                string hostAndPort = Environment.GetEnvironmentVariable(EmulatorHostVariable)?.Trim();
-                string projectId = Environment.GetEnvironmentVariable(EmulatorProjectVariable)?.Trim();
-                var endpoint = string.IsNullOrEmpty(hostAndPort) ? null : hostAndPort;
-                return new EmulatorConfiguration(endpoint, projectId == "" ? null : projectId);
-            }
+            var clientBuilder = new DatastoreClientBuilder(this);
+            clientBuilder.Settings = Settings;
+            return new ConfiguredBuilder(ProjectId, NamespaceId, clientBuilder);
+        }
+
+        private ClientBuilderBase<DatastoreDb> MaybeUseEmulator()
```

Comment:
This method is private, can't you just return DatastoreDbBuilder to save you the cast in the method above?